rest/test/routes/transactionRoutes_spec.js

```diff
@@ -18,96 +18,275 @@
+	});
+
+	describe('transactions', () => {
+		describe('get', () => {
+			const testAddressString = 'SBZ22LWA7GDZLPLQF7PXTMNLWSEZ7ZRVGRMWLXWV';
+			const testAddress = address.stringToAddress(testAddressString);
+
+			const testPublickeyString = '7DE16AEDF57EB9561D3E6EFA4AE66F27ABDA8AEC8BC020B6277360E31619DCE7';
+			const testPublickey = convert.hexToUint8(testPublickeyString);
+
+			const fakeTransaction = { meta: { addresses: [] }, transaction: { type: 12345 } };
+			const dbTransactionsFake = sinon.fake.resolves([fakeTransaction]);
+
+			const mockServer = new MockServer();
+			const db = { transactions: dbTransactionsFake };
+			const services = {
+				config: {
+					pageSize: {
+						min: 10,
+						max: 100,
+						default: 20
+					}
+				}
+			};
+			transactionRoutes.register(mockServer.server, db, services);
+
+			const route = mockServer.getRoute('/transactions').get();
+
+			beforeEach(() => {
+				mockServer.resetStats();
+				dbTransactionsFake.resetHistory();
+			});
+
+			describe('returns transactions', () => {
+				it('returns correct structure with transactions', () => {
+					const req = {
+						params: {}
+					};
+
+					// Act:
+					return mockServer.callRoute(route, req).then(() => {
+						// Assert:
+						expect(mockServer.send.firstCall.args[0]).to.deep.equal({
+							payload: [fakeTransaction],
+							type: 'transactionWithMetadata',
+							structure: 'page'
+						});
+						expect(mockServer.next.calledOnce).to.equal(true);
+					});
+				});
+			});
+
+			describe('parses filters', () => {
+				const runParseFilterTest = (filter, param, value) => {
+					it(filter, () => {
+						const req = { params: { [filter]: param } };
+
+						const expectedResult = {
+							address: undefined,
+							height: undefined,
+							recipientAddress: undefined,
+							signerPublicKey: undefined,
+							state: undefined,
+							transactionTypes: undefined
+						};
+
+						expectedResult[filter] = value;
+
+						// Act + Assert
+						return mockServer.callRoute(route, req).then(() => {
+							expect(dbTransactionsFake.firstCall.args[0]).to.deep.equal(expectedResult);
+						});
+					});
+				};
+
+				const testCases = [
+					{ filter: 'height', param: '15', value: 15 },
+					{ filter: 'address', param: testAddressString, value: testAddress },
+					{ filter: 'signerPublicKey', param: testPublickeyString, value: testPublickey },
+					{ filter: 'recipientAddress', param: testAddressString, value: testAddress },
+					{ filter: 'state', param: 'confirmed', value: 'confirmed' }
+				];
 
-		describe('transactionHash', () => {
-			addGetPostTests(
-				'transactionsByHashes',
-				'transactionId',
-				Valid_Transaction_Hashes,
-				Valid_Transaction_Hashes.map(convert.hexToUint8)
-			);
-			addHomogeneousCheck(Valid_Transaction_Hashes, Valid_Object_Ids[0]);
+				testCases.forEach(testCase => {
+					runParseFilterTest(testCase.filter, testCase.param, testCase.value);
+				});
+
+				it('transactionTypes', () => {
```

Comment:
why is this test standalone and not with the other filters tests above?

Reply:
Because is the only filter where the filter name and the param are different. That made testCases longer and multiline, making it far less readable. It can be added with the rest of the tests adding one more param to runParseFilterTest, but I felt I wasn't necessary. It can be changed though.